src/setup-gcloud.ts

```diff
@@ -100,17 +100,33 @@ export async function run(): Promise<void> {
         credsPath = path.join(githubWorkspace, uniqueName);
       }
 
-      const serviceAccountKeyObj = parseServiceAccountKey(serviceAccountKey);
-      await writeSecureFile(
-        credsPath,
-        JSON.stringify(serviceAccountKeyObj, null, 2), // Print to file as string w/ indents
-      );
+      // If explicit SA input, parse it, write to disk and set GCLOUD_PROJECT.
+      if (serviceAccountKey) {
+        const serviceAccountKeyObj = parseServiceAccountKey(serviceAccountKey);
+        await writeSecureFile(
+          credsPath,
+          JSON.stringify(serviceAccountKeyObj, null, 2), // Print to file as string w/ indents
+        );
+        core.exportVariable('GCLOUD_PROJECT', serviceAccountKeyObj.project_id);
+      } else if (process.env.GOOGLE_GHA_CREDS_PATH) {
```

Comment:
Thinking out loud, why not just throw an exception if `export_default_credentials` is given without `service_account_key`?

Reply:
We could. My reason behind a warning rather than exception was to not break existing workflows if they had `export_default_credentials` when using with auth like https://github.com/google-github-actions/auth/issues/73

Reply:
Makes sense. Can we update the warning text to say "this will be an error in a future release" and then when we cut the next major release we'll make it an error?

Reply:
added